### PR TITLE
TimeSeriesCloudPredictor

### PR DESCRIPTION
**Pull request by yinweisu**
Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/autogluon/cloud/scripts/timeseries_serve.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    elif input_content_type == "application/json":
+        buf = StringIO(request_body)
+        data = pd.read_json(buf)
+
+    elif input_content_type == "application/jsonl":
+        buf = StringIO(request_body)
+        data = pd.read_json(buf, orient="records", lines=True)
+
+    else:
+        raise ValueError(f"{input_content_type} input content type not supported.")
```
[the remark this answers is not in the document]
Comment (yinweisu) [reply src/autogluon/cloud/scripts/timeseries_serve.py:65]: So these are scripts we send to the remote container. There's no way to define functions and share them across these scripts unless we make it in our AG source code

File: src/autogluon/cloud/predictor/timeseries_cloud_predictor.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+    def predict_real_time(
+        self,
+        test_data: Union[str, pd.DataFrame],
+        id_column: str,
+        timestamp_column: str,
+        target: str,
+        static_features: Optional[Union[str, pd.DataFrame]] = None,
+        accept: str = "application/x-parquet",
+    ) -> pd.DataFrame:
```
Comment (austinmw) [inline src/autogluon/cloud/predictor/timeseries_cloud_predictor.py:174]: This is returning a pd.Series

File: src/autogluon/cloud/predictor/timeseries_cloud_predictor.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        """
+        self._validate_predict_real_time_args(accept)
+        test_data = self._preprocess_data(
+            data=test_data,
+            id_column=id_column,
+            timestamp_column=timestamp_column,
+            target=target,
+            static_features=static_features,
+        )
+        pred, _ = self._predict_real_time(test_data=test_data, accept=accept)
```
Comment (austinmw) [inline src/autogluon/cloud/predictor/timeseries_cloud_predictor.py:212]: users typically need mean and quantile forecasts returned

File: src/autogluon/cloud/predictor/multimodal_cloud_predictor.py [hunk 4]
```diff
@@ -163,6 +163,14 @@ def predict(
         **kwargs,
     ) -> Optional[pd.Series]:
         """
+        Predict using SageMaker batch transform.
+        When minimizing latency isn't a concern, then the batch transform functionality may be easier, more scalable, and more appropriate.
+        If you want to minimize latency, use `predict_real_time()` instead.
```
[the remark this answers is not in the document]
Comment (yinweisu) [reply src/autogluon/cloud/predictor/multimodal_cloud_predictor.py:168]: This is mentioned in the predict_real_time api saying that if you want to inference on a large dataset, use batch inference instead.

File: src/autogluon/cloud/predictor/timeseries_cloud_predictor.py (hunk 2 above)
Comment (yinweisu) [reply src/autogluon/cloud/predictor/timeseries_cloud_predictor.py:174]: This is actually a bug. Should be fixed now.

File: src/autogluon/cloud/predictor/timeseries_cloud_predictor.py (hunk 3 above)
Comment (yinweisu) [reply src/autogluon/cloud/predictor/timeseries_cloud_predictor.py:212]: ditto

**Review by tonyhoo — APPROVED** [review]

**Pull request merged**